activemodel/lib/active_model/attribute_methods.rb

```diff
@@ -354,23 +406,16 @@ def initialize(owner, path, line)
             @owner = owner
             @path = path
             @line = line
-            @sources = ["# frozen_string_literal: true\n"]
-            @renames = {}
-          end
-
-          def <<(source_line)
-            @sources << source_line
+            @namespaces = Hash.new { |h, k| h[k] = MethodSet.new(k) }
           end
 
-          def rename_method(old_name, new_name)
-            @renames[old_name] = new_name
+          def define_cached_method(name, namespace:, as: name, &block)
+            @namespaces[namespace].define_cached_method(name, as: as, &block)
```

Comment:
I don't think it's a concern, because that's an internal undocumented API. And if you get it wrong the test suite blows up left and right.

Also getting the backtrace would be quite detrimental for performance (less so on Ruby 3.0 with partial backtraces, but still).